Tests/SonarScanner.MSBuild.TFS.Test/CoverageReportProcessorTests.cs

```diff
@@ -18,151 +18,104 @@
 
     [TestInitialize]
     public void TestInitialize()
     {
-        reportConverter = Substitute.For<ICoverageReportConverter>();
         legacyFactory = Substitute.For<ILegacyTeamBuildFactory>();
-        processor = new CoverageReportProcessor(legacyFactory, reportConverter, new TestLogger());
+        processor = new CoverageReportProcessor(legacyFactory);
     }
 
+    [TestMethod]
     public void Ctor_WhenLegacyFactoryIsNull_Throws()
     {
-        // Arrange
-        Action act = () => new CoverageReportProcessor(null, reportConverter, new TestLogger());
-
-        // Act & Assert
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("legactTeamBuildFactory");
-    }
-
-    public void Ctor_WhenConverterIsNull_Throws()
-    {
-        // Arrange
-        Action act = () => new CoverageReportProcessor(legacyFactory, null, new TestLogger());
-
-        // Act & Assert
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("coverageReportConverter");
-    }
-
-    public void Ctor_WhenLoggerIsNull_Throws()
-    {
-        // Arrange
-        Action act = () => new CoverageReportProcessor(legacyFactory, reportConverter, null);
-
-        // Act & Assert
-        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("logger");
+        Action act = () => _ = new CoverageReportProcessor(null);
+        act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("legacyTeamBuildFactory");
     }
 
     [TestMethod]
     public void Initialize_Checks_Arguments_For_Null()
     {
-        // Arrange
-        var analysisConfig = new AnalysisConfig();
-        Action act = () => processor.Initialize(analysisConfig, null, string.Empty);
-
-        // Act & Assert
+        Action act = () => processor.Initialize(new AnalysisConfig(), null, string.Empty);
         act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("settings");
     }
 
     [TestMethod]
     public void ProcessCoverageReports_LegacyTeamBuild_SkipCoverageIsFalse_WhenProcess_CallsLegacyFactoryThenCallsReturnedProcessor()
     {
-        // Arrange
-        var analysisConfig = new AnalysisConfig { LocalSettings = new AnalysisProperties() };
+        var analysisConfig = new AnalysisConfig { LocalSettings = [] };
         var settingsMock = new MockBuildSettings { BuildEnvironment = BuildEnvironment.LegacyTeamBuild };
         var logger = new TestLogger();
 
         // Set up the factory to return a processor that returns success
-        var processor = Substitute.For<ICoverageReportProcessor>();
-        processor.Initialize(Arg.Any<AnalysisConfig>(), Arg.Any<IBuildSettings>(), Arg.Any<string>()).Returns(true);
-        processor.ProcessCoverageReports(logger).Returns(true);
-        legacyFactory.BuildTfsLegacyCoverageReportProcessor().Returns(processor);
+        var processorSub = Substitute.For<ICoverageReportProcessor>();
```

Comment:
I like the old name more. We typically give it the original name
```suggestion
        var processor = Substitute.For<ICoverageReportProcessor>();
```